cmd/snap-failure/cmd_snapd.go

```diff
@@ -133,14 +133,25 @@ func (c *cmdSnapd) Execute(args []string) error {
 	}
 
 	logger.Noticef("restarting snapd socket")
-	// we need to reset the failure state to be able to restart again
-	if output, err := exec.Command("systemctl", "reset-failed", "snapd.socket").CombinedOutput(); err != nil {
-		return osutil.OutputErr(output, err)
-	}
 	// at this point our manually started snapd stopped and
-	// removed the /run/snap* sockets (this is a feature of
+	// should have removed the /run/snap* sockets (this is a feature of
 	// golang) - we need to restart snapd.socket to make them
 	// available again.
+
+	// we need to reset the failure state to be able to restart again
+	if output, err := exec.Command("systemctl", "reset-failed", "snapd.socket").CombinedOutput(); err != nil {
```

Comment:
Should we `systemctl reset-failed snapd.socket snapd.service` just to be sure?

Reply:
Yes I do this in https://github.com/snapcore/snapd/pull/8169, I'd rather land this as-is and do more followups there